docker: Add build file

rzr opened the pull request:
This can be useful for building from git,

For example on odroid-xu4, I was able to build:

    webthings-gateway_0.11.0-1_armhf-debian-buster.deb

Using docker:

    docker build -t gateway-deb \
      https://github.com/CrossStream/gateway-deb.git

Relate-to: https://twitter.com/RzrFreeFr/status/1230918703107190784
Change-Id: I4dfad531198fff9f37f04c429c01aee0edcc6b5b
Signed-off-by: Philippe Coval <rzr@users.sf.net>

rzr committed:
docker: Add build file

This can be useful for building from git,

For example on odroid-xu4, I was able to build:

    webthings-gateway_0.11.0-1_armhf-debian-buster.deb

Using docker:

    docker build -t gateway-deb \
      https://github.com/CrossStream/gateway-deb.git

Relate-to: https://twitter.com/RzrFreeFr/status/1230918703107190784
Forwarded: https://github.com/mozilla-iot/gateway-deb/pull/6
Change-Id: I4dfad531198fff9f37f04c429c01aee0edcc6b5b
Signed-off-by: Philippe Coval <rzr@users.sf.net>

mrstegeman commented:
As far as I can tell, there's no predefined way to get the package out of the container. Am I missing something? Maybe there should be a `build-docker.sh` script that runs the container and copies the `.deb` out? It would essentially end up doing this: https://github.com/mozilla-iot/gateway-deb/blob/master/.github/workflows/pr.yml#L50-L53

rzr commented:
Well adding a script would require a clone by user.
It's easy to pull the deb file out , 

I have something like this in my scripts:

    tag="latest"
    repository="gateway-deb"
    image="$repository:$tag"
    container=$(docker create "$image")
    workdir=/usr/local/opt/${project}/src/${project}
    file=${workdir}/webthings-gateway_0.11.0-1_armhf-debian-buster.deb
    dst=/var/lib/docker/local/tmp/
    mkdir -p "$dst"
    docker cp "$container:$file" "$dst"

I can add a script in later change (for 0.12.0)

Then I will also add a change for gh matrix...

Stay tunned

rzr committed:
debian: Add lsb as dep

It's used by scripts and not shipped in dockerhub images

Forwarded: https://github.com/mozilla-iot/gateway-deb/pull/6
Change-Id: I1e67938f3f08a7b0170ad21c985c21bd115c2907
Signed-off-by: Philippe Coval <rzr@users.sf.net>

rzr committed:
docker: Add build script to pull packages out of container

Change-Id: I4d053e771d880001a5adaf503006fc1a409cbe30
Forwarded: https://github.com/mozilla-iot/gateway-deb/pull/6
Signed-off-by: Philippe Coval <rzr@users.sf.net>

rzr committed:
docker: Add docker-compose.yml

To run gateway in docker container

Forwarded: https://github.com/mozilla-iot/gateway-deb/pull/6
Change-Id: I301af1038c518ee88c58a001057e748e8b880af3
Signed-off-by: Philippe Coval <rzr@users.sf.net>
